core-rust/state-manager/src/store/address_book_components.rs

```diff
@@ -0,0 +1,40 @@
+use crate::engine_prelude::Sbor;
+
+
+/// The ID of the node stored in the Address book (Secp256k1 public key)
+#[derive(Clone, Copy, Sbor)]
+pub struct AddressBookNodeId(pub [u8; AddressBookNodeId::LENGTH]);
```

Comment:
This stuff shouldn't be in the `state-manager` crate as it's not relevant to the ledger / state manager database.

We may want to introduce new crates for `consensus` to stick the safety store, and `p2p` for the address book?

Where `jni-export` includes `consensus`, `p2p` etc.

We may then want to move some of the ledger helpers into `node-common`.

Reply:
Makes sense. Will refactor these.